### PR TITLE
Support Device Classes

### PR DESCRIPTION
**Pull request by rklomp**
This is an untested change proposal for https://github.com/home-assistant/core/issues/49592

### CODE REVIEW

**Comment by rklomp** [comment]
Tested this in my home assistant instance and still working fine.
I have asked @itinside to test it as well to see it it fixes https://github.com/home-assistant/core/issues/49592

File: pysma/__init__.py [hunk 1]
```diff
@@ -18,8 +18,7 @@
 USERS = {"user": "usr", "installer": "istl"}
 
 JMESPATH_BASE = "result.*"
-JMESPATH_VAL_IDX = '"1"[{}].val'
-JMESPATH_VAL = "val"
+JMESPATH_VAL = ["val", '"1"[{}].val', '"65"[{}].val']
```
Comment (kellerza) [inline pysma/__init__.py:21]:
```suggestion
JMESPATH_VAL = ["val", '"1"[{}].val', '"*"[{}].val']
```

**Comment by rklomp** [comment]
Just found this in de javascript of the webinterface:

```javascript
    DevClassInverter: "1",
    DevClassBattery: "7",
    DevClassEnergyMeter: "65"
```

Looks like these are the options.

And:

```javascript
this.getDevClassAndValuesArrayFromValueObject = function(a) {
        if (angular.isObject(a)) {
            var b = Object.keys(a);
            if (1 !== b.length) return void c.error(a, "has more or less than one device class. Not supported yet!");
            var d = b[0];
            return {
                devClass: d,
                valuesArray: a[d]
            }
        }
```

Looks like even the web interface only supports one device class.

**Comment by kellerza** [comment]
Great, so `val` and then wildcard `*[{}]val` should be ok as well.

**Comment by rklomp** [comment]
I will have a look if I can create a similar function in `pysma` and extract the device class to inject that in the JMESPATH_VAL.
It might be usefull later to only load the sensors specific to that device class in HA.

**Comment by rklomp** [comment]
@kellerza , How about this. This will extract the devclass on first read and uses the stored value on subsequent reads.

After this is merged I could create a sensor map that maps certain sensors to specific device classes. For example the sensors mentioned in https://github.com/kellerza/pysma/issues/58 will only be used for devclass == 7

File: pysma/__init__.py [hunk 2]
```diff
@@ -426,11 +431,22 @@ async def device_info(self):
...
             or fallback_device_info["type"],
             "manufacturer": self.device_info_sensors["device_manufacturer"].value
             or fallback_device_info["manufacturer"],
         }
 
         return device_info
+
+    async def get_devclass(self, result_body):
+        if not self.devclass:
+            sensor_value = list(result_body.values())[0]
```
Comment (kellerza) [inline pysma/__init__.py:445]: What if the result_body was empty? Then you cannot access [0]?

Comment (rklomp) [reply pysma/__init__.py]: You are right, I have updated it to return None in that case.

**Comment by rklomp** [comment]
Depending whether multiple devices classes are served from the same device, we might need a different approach.
Tring to figure that out in https://github.com/kellerza/pysma/issues/58

**Pull request merged**